### PR TITLE
feat(views-students): [skp-132] create student query endpoint getAll()

### PR DESCRIPTION
**Pull request by aurimas12**

### CODE REVIEW

File: apps/api/src/app/view/students/queries/get-students.handlers.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { Inject } from '@nestjs/common';
+import { StudentModelService } from '../../../models/students/student.service';
+
+export class GetStudentsQuery { }
+
+@QueryHandler(GetStudentsQuery)
+export class GetStudentsHandler implements IQueryHandler<GetStudentsQuery>{
+    @Inject() private readonly _studentModelService: StudentModelService;
+    async execute() {
+        return this._studentModelService.model.find().exec();
```
Comment (Gardemarin) [inline apps/api/src/app/view/students/queries/get-students.handlers.ts:11]: Replace `.model.find().exec();` to getAll(), create `getAll(){}` function inside StundentModelService and return `    return this._model.find().exec();`

File: apps/api/src/app/models/students/dto/student-payload.dto.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    @ApiProperty()
+    public readonly lastName: string;
+    @ApiProperty()
+    public readonly photo: string;
+    @ApiProperty()
+    public readonly fieldOfProfession: string;
+    @ApiProperty()
+    public readonly company: string;
+    @ApiProperty()
+    public readonly categories: string;
```
Comment (Gardemarin) [inline apps/api/src/app/models/students/dto/student-payload.dto.ts:20]: `Categories` is plural, so why is it string only?

File: apps/api/src/app/models/students/dto/create-student-payload.dto.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export class CreateStudentPayloadDto implements CreateStudent {
     @ApiProperty()
     public readonly company: string;
     @ApiProperty()
-    public readonly categories: string;
+    public readonly categories: [String];
```
Comment (doncevas) [inline apps/api/src/app/models/students/dto/create-student-payload.dto.ts:20]: string[]

File: apps/api/src/app/models/students/dto/create-student-payload.dto.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@ export class CreateStudentPayloadDto implements CreateStudent {
     @ApiProperty()
     public readonly company: string;
     @ApiProperty()
```
Comment (doncevas) [inline apps/api/src/app/models/students/dto/create-student-payload.dto.ts:19]: @ApiProperty({isArray: true, type: 'string'})

File: apps/api/src/app/models/students/dto/student-payload.dto.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { CreateStudent } from '../student.interface';
+
+export class StudentPayloadDto implements CreateStudent {
```
Comment (doncevas) [inline apps/api/src/app/models/students/dto/student-payload.dto.ts:4]: don't duplicate dtos

File: apps/api/src/app/models/students/student.interface.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ export interface CreateStudent {
   photo: string,
   fieldOfProfession: string,
   company: string,
-  categories: string,
+  categories: [String],
```
Comment (doncevas) [inline apps/api/src/app/models/students/student.interface.ts:11]: string[]

File: apps/api/src/app/models/students/student.interface.ts [hunk 7]
```diff
@@ -22,7 +22,7 @@ export interface UpdateStudent {
...
```
Comment (doncevas) [inline apps/api/src/app/models/students/student.interface.ts:25]: string[]

File: apps/api/src/app/models/students/dto/update-student-payload.dto copy.ts [hunk 8]
```diff
@@ -17,7 +17,7 @@ export class UpdateStudentPayloadDto implements UpdateStudent {
     @ApiProperty()
     public readonly company: string;
     @ApiProperty()
-    public readonly categories: string;
+    public readonly categories: [string];
```
Comment (doncevas) [inline apps/api/src/app/models/students/dto/update-student-payload.dto copy.ts:20]: string[]

File: apps/api/src/app/models/students/student.interface.ts [hunk 9]
```diff
@@ -1,14 +1,25 @@
 import { Document } from 'mongoose';
 
 export interface CreateStudent {
+
+  firstName?: string,
+  lastName?: string,
+  photo?: string,
+  fieldOfProfession?: string,
+
+  categories?: [string],
```
Comment (doncevas) [inline apps/api/src/app/models/students/student.interface.ts:10]: string[]

File: apps/api/src/app/models/students/student.interface.ts [hunk 10]
```diff
@@ -1,14 +1,25 @@
...
+export interface Student {
   type: string,
   description: string,
   firstName: string,
   lastName: string,
   photo: string,
   fieldOfProfession: string,
   company: string,
-  categories: string,
+  categories: [string],
```
Comment (doncevas) [inline apps/api/src/app/models/students/student.interface.ts:22]: string[]

File: apps/api/src/app/domains/students/students.controller.ts [hunk 11]
```diff
@@ -11,9 +11,9 @@ import { UpdateStudentPayloadDto } from '../../models/students/dto/update-studen
 import { UpdateStudentCommand } from './commands/update-student.command';
 
 @Controller('students')
-@ApiBearerAuth('JWT')
-@UseGuards(AuthGuard('jwt'))
-@UseInterceptors(LoggingInterceptor, TransformInterceptor)
+// @ApiBearerAuth('JWT')
+// @UseGuards(AuthGuard('jwt'))
+// @UseInterceptors(LoggingInterceptor, TransformInterceptor)
```
Comment (doncevas) [inline apps/api/src/app/domains/students/students.controller.ts:16]: revert

**Pull request merged**